src/components/EmptyStateComponent/index.tsx

```diff
@@ -0,0 +1,99 @@
+import type {VideoReadyForDisplayEvent} from 'expo-av';
+import React, {useMemo, useState} from 'react';
+import {View} from 'react-native';
+import Button from '@components/Button';
+import ImageSVG from '@components/ImageSVG';
+import Lottie from '@components/Lottie';
+import Text from '@components/Text';
+import VideoPlayer from '@components/VideoPlayer';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import CONST from '@src/CONST';
+import type {EmptyStateComponentProps, VideoLoadedEventType} from './types';
+
+const VIDEO_ASPECT_RATIO = 400 / 225;
+
+function EmptyStateComponent({SkeletonComponent, headerMediaType, headerMedia, buttonText, buttonAction, title, subtitle, headerStyles, headerContentStyles}: EmptyStateComponentProps) {
+    const styles = useThemeStyles();
+    const {isSmallScreenWidth} = useWindowDimensions();
+
+    const [videoAspectRatio, setVideoAspectRatio] = useState(VIDEO_ASPECT_RATIO);
+
+    const setAspectRatio = (event: VideoReadyForDisplayEvent | VideoLoadedEventType | undefined) => {
+        if (!event) {
+            return;
+        }
+
+        if ('naturalSize' in event) {
+            setVideoAspectRatio(event.naturalSize.width / event.naturalSize.height);
+        } else {
+            setVideoAspectRatio(event.srcElement.videoWidth / event.srcElement.videoHeight);
+        }
+    };
+
+    const HeaderComponent = useMemo(() => {
+        switch (headerMediaType) {
+            case 'video':
```

Comment:
We can move 'video',  'illustration',  'animation' strings to `CONST`, then use them here and in `src/components/EmptyStateComponent/types.ts`

Reply:
Done